__e2e__/root.test.ts

```diff
@@ -0,0 +1,62 @@
+import path from 'path';
+
+import {
+  spawnScript,
+  runCli,
+  getTempDirectory,
+  cleanup,
+  writeFiles,
+} from '../jest/helpers';
+
+const cwd = getTempDirectory('test_different_roots');
+
+beforeAll(() => {
+  // Register all packages to be linked
+  for (const pkg of ['platform-ios', 'platform-android']) {
+    spawnScript('yarn', ['link'], {
+      cwd: path.join(__dirname, `../packages/${pkg}`),
+    });
+  }
+
+  // Clean up folder and re-create a new project
+  cleanup(cwd);
+  writeFiles(cwd, {});
+
+  // Initialise React Native project
+  runCli(cwd, ['init', 'TestProject']);
+
+  // Link CLI to the project
+  for (const pkg of [
+    '@react-native-community/cli-platform-ios',
+    '@react-native-community/cli-platform-android',
+  ]) {
+    spawnScript('yarn', ['link', pkg], {
+      cwd: path.join(cwd, 'TestProject'),
+    });
+  }
+});
+
+afterAll(() => {
+  cleanup(cwd);
+});
+
+test('works when Gradle is run outside of the project hierarchy', () => {
+  /**
+   * Location of Android project
+   */
+  const androidProjectRoot = path.join(cwd, 'TestProject/android');
+
+  /*
+   * Grab absolute path to Gradle wrapper. The fact that we are using
+   * a wrapper from the project is just a convinience to avoid installing
+   * Gradle globally
+   */
+  const gradleWrapper = path.join(androidProjectRoot, 'gradlew');
+
+  // Execute `gradle` with `-p` flag and `cwd` outside of project hierarchy
+  const {stdout} = spawnScript(gradleWrapper, ['-p', androidProjectRoot], {
+    cwd: '/',
```

Comment:
Do we need to set cwd here? I'd expect a default value would be sufficient – this way we could make the `options` argument of `spawnScript` optional with `{}` as default value, as it used to be

Reply:
Truth is - we never want to rely on `default` value in our tests and that's why I decided to go with explicit `cwd`. 

In this case, I just want to make sure and be 100% explicit that `cwd` is somewhere else than the project - for example, at the root.

Reply:
> Truth is - we never want to rely on default value in our tests and that's why I decided to go with explicit cwd.

The reason why I believe it's better to require `cwd` is simple - `cwd` defaults to the location where the React Native CLI source files are located (or where the tests are being run from).

Conceptually, I think of `e2e` tests as something run in a totally separate environment that is close to what users are doing in real life. I feel like setting `cwd` in `spawnScript` (which I consider advanced use case, where `runCLI` is not enough) is just a way to tell "watch out, you're going into advanced territory - think about what you are doing".